tests/HashMapProperties.hs

```diff
@@ -351,6 +351,12 @@ pFromListWith :: [(Key, Int)] -> Bool
 pFromListWith kvs = (M.toAscList $ M.fromListWith (+) kvs) ==
                     (toAscList $ HM.fromListWith (+) kvs)
 
+pFromListWithKey :: [(Key, Int)] -> Bool
+pFromListWithKey kvs = (M.toAscList $ M.fromListWithKey combine kvs) ==
+                       (toAscList $ HM.fromListWithKey combine kvs)
+  -- Ensure that the way values are combined depend on the key
+  where combine (K k) v1 v2 = (v1 + v2) `mod` (abs k + 1)
```

Comment:
I don't find it easy to understand this test. I think `combine` might be commutative though, so it might fail to catch some errors, no?

Reply:
I agree it's a bit contrived. But it's not that much different from the test above it, it's just the `combine` function which mixes in the key in the computation.

As for commutativity, the `pFromListWith` uses commutative operators and there's where I got the inspiration. I suppose I could use e.g. subtraction instead.

Reply:
Yeah, feel free to improve `pFromListWith` too! Subtraction is both non-commutative _and_ non-associative, which seems ideal!

Reply:
Done. Thanks.